src/common/snippets/src/pass/gn_tokenization.cpp

```diff
@@ -0,0 +1,38 @@
+// Copyright (C) 2023 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "snippets/pass/gn_tokenization.hpp"
+#include "snippets/pass/collapse_subgraph.hpp"
+
+#include "snippets/itt.hpp"
+#include "snippets/op/subgraph.hpp"
+#include "snippets/utils.hpp"
+
+#include "openvino/core/rt_info.hpp"
+#include "openvino/pass/pattern/op/wrap_type.hpp"
+
+ov::snippets::pass::TokenizeGNSnippets::TokenizeGNSnippets() {
+    MATCHER_SCOPE(TokenizeGNSnippets);
+
+    auto group_norm_pattern = ov::pass::pattern::wrap_type<ov::op::v12::GroupNormalization>();
+
+    ov::matcher_pass_callback callback = [=](ov::pass::pattern::Matcher& m) {
+        OV_ITT_SCOPED_TASK(ov::pass::itt::domains::SnippetsTransform, "Snippets::pass::TokenizeGNSnippets")
+        auto group_norm_node = ov::as_type_ptr<ov::op::v12::GroupNormalization>(m.get_match_root());
+        if (group_norm_node->is_dynamic() || group_norm_node->get_element_type() != element::f32)
+            return false;
+
+        auto subgraph = op::Subgraph::wrap_node_as_subgraph(group_norm_node);
+        subgraph->get_rt_info()["originalLayersNames"] = group_norm_node->get_friendly_name();
+        ov::replace_node(group_norm_node, subgraph);
+        op::update_out_tensor_name(subgraph);
+
+        // mark the Subgraph as Completed to not allow Snippets to include any nodes into the GN Subgraph in common Tokenization
+        SetSnippetsSubgraphType(subgraph, SnippetsSubgraphType::Completed);
```

Comment:
Could you please explain (here and in the code) why can't we append operations to the GN subgraph. What pipeline limitations impose this restriction? How can we resolve/relax these limitations?

Reply:
The major reason is GN subgraph scheduling should be fixed as parallel for batch * group. This is not optimal for other part subgraph, which could introduce regression.

Reply:
Could you please create a ticket for this and explain the issue in detail there (why is this suboptimal, is the root cause specific to this opreation only, how can the limitation be resolved, attach graphs that illustrate the particular problem)? And also add the ticket number to the comment.
This is very important for us to understand current limitations (and possible ways to fix them) in order to prioritize future activities in an optimal way. Please share this info 🙂

Reply:
Ticket is create and added to code comment. Thanks Ivan!

Reply:
> Could you please create a ticket for this and explain the issue in detail there (why is this suboptimal, is the root cause specific to this opreation only, how can the limitation be resolved, attach graphs that illustrate the particular problem)? And also add the ticket number to the comment. This is very important for us to understand current limitations (and possible ways to fix them) in order to prioritize future activities in an optimal way. Please share this info 🙂

tickets number is 137310 :-)